keyboards/geekboards/macropad_v2/config.h

```diff
@@ -0,0 +1,94 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x0483
+#define PRODUCT_ID      0xA372
+#define DEVICE_VER      0x0002
+#define MANUFACTURER    GeekBoards
+#define PRODUCT         GeekBoards Macropad v2
+#define DESCRIPTION     GeekBoards Macropad v2
+
+#define USB_MAX_POWER_CONSUMPTION 500
+
+/* key matrix size */
+#define MATRIX_ROWS 2
+#define MATRIX_COLS 4
+
+#define DIRECT_PINS {{B13, B15, B3, B5}, {B12, B14, A13, B7}}
+#define UNUSED_PINS
+
+#define RGBLED_NUM 42
+#define DRIVER_LED_TOTAL 42
+#define RGB_DI_PIN A7
+
+// PWM RGB Underglow Defines
+#define WS2812_PWM_DRIVER PWMD3
+#define WS2812_PWM_CHANNEL 2
+#define WS2812_PWM_PAL_MODE 1
+#define WS2812_DMA_STREAM STM32_DMA1_STREAM3
+#define WS2812_DMA_CHANNEL 3
+
+#ifdef RGB_MATRIX_ENABLE
+#define RGB_MATRIX_KEYPRESSES 
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#define DISABLE_RGB_MATRIX_ALPHAS_MODS
+#define DISABLE_RGB_MATRIX_BAND_SAT
+#define DISABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
+#define DISABLE_RGB_MATRIX_BAND_PINWHEEL_VAL
+#define DISABLE_RGB_MATRIX_BAND_SPIRAL_SAT
+#define DISABLE_RGB_MATRIX_BAND_SPIRAL_VAL
+#define DISABLE_RGB_MATRIX_CYCLE_OUT_IN
+#define DISABLE_RGB_MATRIX_RAINBOW_MOVING_CHEVRON
+#define DISABLE_RGB_MATRIX_CYCLE_SPIRAL
+#define DISABLE_RGB_MATRIX_RAINBOW_BEACON
+#define DISABLE_RGB_MATRIX_RAINBOW_PINWHEELS
+#define DISABLE_RGB_MATRIX_RAINDROPS
+#define DISABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
+#define DISABLE_RGB_MATRIX_DIGITAL_RAIN
+#define DISABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE
+#define DISABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
+#define DISABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
+
+#define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_CYCLE_UP_DOWN
+#define RGB_MATRIX_STARTUP_SAT 255
+#define RGB_MATRIX_STARTUP_VAL 192
+#define RGB_MATRIX_STARTUP_SPD 30
+#endif //RGB_MATRIX_ENABLE
+
+#define RGB_DISABLE_WHEN_USB_SUSPENDED true
+#define WAIT_FOR_USB
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCING_DELAY 3
+
+/* Bootmagic Lite key configuration */
+#define BOOTMAGIC_LITE_ROW 0
+#define BOOTMAGIC_LITE_COLUMN 0
+
+#define PERMISSIVE_HOLD
+#undef TAPPING_TOGGLE
+#define TAPPING_TOGGLE  3
+#define RETRO_TAPPING
+
+#ifndef __ASSEMBLER__
+void eeconfig_update_rgb_matrix(void);
+#define eeconfig_update_rgblight_current eeconfig_update_rgb_matrix
+#endif
```

Comment:
```suggestion
```

Reply:
This is necessary to use RGB_MATRIX with VIA, and this is the only way it works, that don't require core changes. I discussed this a couple times on discord, and nobody supported my idea to include these changes into core. This also requires a bit of wizardry in VIA config, nothing criminal though. Here is a demo: https://youtu.be/Sy0ZG9p_KW0

Reply:
If found appropriate, I would be happy to PR this into core, it is much more straigtforward, just a couple lines added in rgb_matrix.h.